src/clj/fluree/db/reasoner/core.cljc

```diff
@@ -0,0 +1,156 @@
+(ns fluree.db.reasoner.core
+  (:require [clojure.core.async :as async :refer [alts! go]]
+            [fluree.db.flake :as flake]
+            [fluree.db.util.core :as util]
+            [fluree.db.util.log :as log]
+            [fluree.db.util.async :refer [go-try <?]]
+            [fluree.db.reasoner.resolve :as resolve]
+            [fluree.db.json-ld.transact :as transact]
+            [fluree.db.fuel :as fuel]
+            [fluree.json-ld :as json-ld]
+            [fluree.db.reasoner.graph :refer [task-queue add-rule-dependencies]]))
+
+#?(:clj (set! *warn-on-reflection* true))
+
+(defn reasoned-flake?
+  "Returns truthy if the flake has been generated by reasoner"
+  [flake]
+  (-> flake flake/p :reasoned))
```

Comment:
Should this be `flake/m` instead of `flake/p`? It looks like when we create flakes we store this key in meta.

Reply:
Yes! Thanks.